flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlStateFactory.java

```diff
@@ -126,15 +128,15 @@ private IS createState() throws Exception {
 	@SuppressWarnings("unchecked")
 	private IS createValueState() throws Exception {
 		ValueStateDescriptor<TtlValue<SV>> ttlDescriptor = new ValueStateDescriptor<>(
-			stateDesc.getName(), new TtlSerializer<>(stateDesc.getSerializer()));
+			stateDesc.getName(), new TtlSerializer<>(LongSerializer.INSTANCE, stateDesc.getSerializer()));
```

Comment:
As mentioned above, having to pass in a `LongSerializer.INSTANCE` every time we're instantiating a TtlSerializer seems very redundant.

Reply:
I disagree, same reasons as above.